packages/performance/src/definitions.ts

```diff
@@ -105,3 +147,176 @@ export interface IsEnabledResult {
    */
   enabled: boolean;
 }
+
+/**
+ * @since 6.3.0
+ */
+export interface PutAttributeOptions {
+  /**
+   * Name of the trace to set its attribute.
+   *
+   * @since 6.3.0
+   */
+  traceName: string;
+  /**
+   * Name of the attribute to set its value.
+   *
+   * @since 6.3.0
+   * @example "experiment"
+   */
+  attribute: string;
+  /**
+   * The value to set to the attribute.
+   *
+   * @since 6.3.0
+   * @example "A"
+   */
+  value: string;
+}
+
+/**
+ * @since 6.3.0
+ */
+export interface GetAttributeOptions {
+  /**
+   * Name of the trace to set its attribute.
+   *
+   * @since 6.3.0
+   */
+  traceName: string;
+  /**
+   * Name of the attribute to retrieve its value.
+   *
+   * @since 6.3.0
+   */
+  attribute: string;
+}
+
+/**
+ * @since 6.3.0
+ */
+export interface GetAttributeResult {
+  /**
+   * The value of the custom attribute.
+   *
+   * @since 6.3.0
+   */
+  value: string | null;
+}
+
+/**
+ * @since 6.3.0
+ */
+export interface GetAttributesOptions {
+  /**
+   * Name of the trace to get its attributes.
+   *
+   * @since 6.3.0
+   */
+  traceName: string;
+}
+
+/**
+ * @since 6.3.0
+ */
+export interface GetAttributesResult {
+  /**
+   * A map of all custom attributes of a trace with their values.
+   *
+   * @since 6.3.0
+   */
+  result: { [key: string]: string };
+}
+
+/**
+ * @since 6.3.0
+ */
+export type RemoveAttributeOptions = GetAttributeOptions;
+
+/**
+ * @since 6.3.0
+ */
+export interface PutMetricOptions {
+  /**
+   * Name of the trace to set its metric.
+   *
+   * @since 6.3.0
+   */
+  traceName: string;
+  /**
+   * The metric name.
+   *
+   * @since 6.3.0
+   */
+  metricName: string;
+  /**
+   * The value to set for the metric.
+   * The given value is floored down to the nearest integer.
+   *
+   * @since 6.3.0
+   */
+  num: number;
+}
+
+/**
+ * @since 6.3.0
+ */
+export interface GetMetricOptions {
+  /**
+   * Name of the trace to get its metric.
+   *
+   * @since 6.3.0
+   */
+  traceName: string;
+  /**
+   * The metric name.
+   *
+   * @since 6.3.0
+   */
+  metricName: string;
+}
+
+/**
+ * @since 6.3.0
+ */
+export interface GetMetricResult {
+  /**
+   * The value of the metric if exists.
+   *
+   * @since 6.3.0
+   */
+  value: number | null;
```

Comment:
```suggestion
  /**
   * The value of the metric. If a custom metric with that name does not exist will return zero.
   *
   * @since 6.3.0
   */
  value: number;
```

We should not change the behavior of the Firebase SDKs. These plugins are just wrappers.